src/hardware/IOSwitch.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file IOSwitch.h
+ * 
+ * @brief A physical switch connected to a GPIO Pin  
+ */
+
+#include "GPIOPin.h"
+#include "IOSwitch.h"
+
+IOSwitch::IOSwitch(GPIOPin& gpioInstance, SwitchType type)
+    : gpio(gpioInstance),
+      switchType(type),
+      debounceDelay(20),
+      longPressDuration(500),
+      lastState(LOW),
+      currentState(LOW),
+      lastDebounceTime(0),
+      lastStateChangeTime(0),
+      pressStartTime(0),
+      longPressTriggered(false) { 
+}
+
+bool IOSwitch::isPressed() {
+    uint8_t reading = gpio.read();
+    unsigned long currentMillis = millis();
+
+    if (reading != lastState) {
+        lastDebounceTime = currentMillis;
+    }
+
+    if ((currentMillis - lastDebounceTime) > debounceDelay) {
+        if (reading != currentState) {
+            currentState = reading;
+
+            if (currentState == LOW) {
```

Comment:
This logic seems flawed to me, the condition will never occur since you *just* set `currentState = reading`.

Reply:
I misread. This is fine.